src/app/api/users/route.ts

```diff
@@ -0,0 +1,27 @@
+import { user } from "@/dao/user.dao";
+import { NextApiRequest } from "next";
+import { NextResponse } from "next/server";
+
+export async function GET(req: NextApiRequest) {
+  try {
+    const { username, email, page, pageSize } = req.query || {};
+
+    const filter = {
+      username: typeof username === "string" ? username : undefined,
+      email: typeof email === "string" ? email : undefined,
```

Comment:
we should not be asserting types in TypeScript. The comment above should solve this